src/app/api/categories/[categoryId]/route.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  response200,
+  response400,
+  response404,
+  response500,
+} from "@/app/utils";
+import { getProductsByCategory } from "@/model/ProductModel";
+
+export async function GET(
+  request: Request,
+  {
+    params,
+  }: {
+    params: { categoryId: string };
+  }
+) {
+  try {
+    let products = await getProductsByCategory(params.categoryId);
+    if (products !== undefined) {
```

Comment:
Why are you checking against `undefined` here?

The `getProductsByCategory` function returns a `Promise<Product[]>` so this `if` doesn't make sense.

---

What you should do is:

1. Get the category
2. If the category doesn't exist, return `404`
3. If it does, get the products and return them